plugins/rest/rest_test.go

```diff
@@ -543,6 +543,55 @@ func TestNew(t *testing.T) {
 				}
 			}`, grantTypeClientCredentials),
 		},
+		{
```

Comment:
@prasanthu we should have a test that exercises the workflow. There are examples of this in `plugins/rest/rest_test.go`.  WDYT?

Reply:
I had a look at the `TestOauth2ClientCredentialsJwtAuthentication`. The current setup won't allow to mock the client_assertion creation.

Refer these methods:
https://github.com/open-policy-agent/opa/blob/main/plugins/rest/rest_test.go#L1632
https://github.com/open-policy-agent/opa/blob/main/plugins/rest/rest_test.go#L2002

Any pointers on how to mock the generation of client_assertion

Reply:
Can you help me understand where you would like to have the ability to mock the generation of client_assertion? We should be able to customize the test to accommodate this I would imagine.

Reply:
When we use KMS to sign the assertion, the `SignDigest()` gets invoked. It will send a signed AWS request to  the KMS endpoint (based on the environment) for signing.

Reply:
Ok. Unless I'm missing something the test you mentioned creates a test server which `SignDigest` calls and then the test verifies the signature. We should be able to add a new similar test, correct?

Reply:
I have added a test which implements your suggestion. Can you check `TestOauth2ClientCredentialsGrantTypeWithKms`